libs/ledger/tests/chain/main_chain_service_tests.cpp

```diff
@@ -474,14 +628,147 @@ TEST_F(MainChainServiceTests, CheckWhenGenesisAppearsToBeInvalid)
   log.status = fetch::ledger::TravelogueStatus::NOT_FOUND;
 
   EXPECT_CALL(endpoint_, GetDirectlyConnectedPeers()).WillRepeatedly(Return(AddressList{other1_}));
-  EXPECT_CALL(rpc_client_, TimeTravel(other1_, GetGenesisDigest()))
+  EXPECT_CALL(rpc_client_, TimeTravel(other1_, ExpectedHash(GetGenesisDigest())))
       .WillOnce(Return(CreatePromise(log)));
 
   Tick(State::SYNCHRONISING, State::START_SYNC_WITH_PEER);
   Tick(State::START_SYNC_WITH_PEER, State::REQUEST_NEXT_BLOCKS);
   Tick(State::REQUEST_NEXT_BLOCKS, State::WAIT_FOR_NEXT_BLOCKS);
-  Tick(State::WAIT_FOR_NEXT_BLOCKS, State::REQUEST_NEXT_BLOCKS);
-  Tick(State::REQUEST_NEXT_BLOCKS, State::COMPLETE_SYNC_WITH_PEER);
+  Tick(State::WAIT_FOR_NEXT_BLOCKS, State::COMPLETE_SYNC_WITH_PEER);
+}
+
+namespace {
+
+MainChainProtocol::Travelogue TimeTravel(BlockGenerator::BlockPtr const &heaviest_block,
+                                         BlockGenerator::BlockPtrs       local_blocks)
+{
+  return {heaviest_block->hash, heaviest_block->block_number, TravelogueStatus::HEAVIEST_BRANCH,
+          std::move(local_blocks)};
+}
+
+}  // namespace
+
+TEST_F(MainChainServiceTests, CheckExponentialBackStep)
```

Comment:
How long does this test take to run, does it make sense to either move it or make it part of part of the integration test suite?